### PR TITLE
Feature tests

### PR DESCRIPTION
**Pull request by bpmweel**
Implementation of more unit tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,4 +12,7 @@ __pycache__
 .pytest_cache/
 satsense.egg-info/
 test-reports/
+.coverage*
```
Comment (bouweandela) [inline .gitignore]: These are duplicates

File: test/data/source/section_2_sentinel.tif.aux.xml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+<PAMDataset>
```
Comment (bouweandela) [inline test/data/source/section_2_sentinel.tif.aux.xml]: Did you want to include these xml files or are they here accidentally?

File: .gitignore (hunk 1 above)
Comment (bpmweel) [reply .gitignore]: Fixed with 34abe54

File: test/data/source/section_2_sentinel.tif.aux.xml (hunk 2 above)
Comment (bpmweel) [reply test/data/source/section_2_sentinel.tif.aux.xml]: Fixed with 34abe54

**Comment by bouweandela** [comment]
Maybe it would be nice to use [parametrized fixtures](https://docs.pytest.org/en/latest/fixture.html#parametrizing-fixtures) or use [`pytest.mark.parametrize`](https://docs.pytest.org/en/latest/parametrize.html) to reduce the amount of duplicated code in the test functions.

**Pull request merged**